File: 2016/12/12.py
def generator(input): 
    return [line.split() for line in input.splitlines()]
    
def part_1(program): 
    return execute(program, 1)
    
def part_2(program): 
    return execute(program, 8)
    
def execute(program, delta):
    fibo = int(program[2][1])
    fa   = int(program[16][1])
    fb   = int(program[17][1])
    a, b = 1, 1

    for _ in range(fibo + delta): 
        a, b = b, a + b
    return a + fa * fb

File: 2016/12/12.rs
aoc::main!();

fn preprocessing(input: &str) -> Vec<usize> {
    [2, 16, 17]
    .iter()
    .map(|&n| input.lines().nth(n).unwrap())
    .map(|l| l.split(' ').nth(1).unwrap().parse::<usize>().unwrap())
    .collect_vec()
}

fn part_1(program: Vec<usize>) -> usize {
    solver(program, 1)
}

fn part_2(program: Vec<usize>) -> usize {
    solver(program, 8)
}


fn solver(program: Vec<usize>, delta: usize) -> usize {
    (0..program[0] + delta)
    .fold((1,1), |acc, _| (acc.1, acc.0 + acc.1))
    .0
    + program[1] * program[2]
}